griptape/config/base_structure_config.py

```diff
@@ -36,6 +38,26 @@ class BaseStructureConfig(BaseConfig, ABC):
     text_to_speech_driver: BaseTextToSpeechDriver = field(kw_only=True, metadata={"serializable": True})
     audio_transcription_driver: BaseAudioTranscriptionDriver = field(kw_only=True, metadata={"serializable": True})
 
+    _structure: Optional[Structure] = field(default=None, kw_only=True, alias="structure")
+
+    @property
+    def structure(self) -> Optional[Structure]:
+        return self._structure
+
+    @structure.setter
+    def structure(self, structure: Structure) -> None:
+        self._structure = structure
+
+        event_listener = EventListener(self.structure.publish_event)
+
+        self.prompt_driver.add_event_listener(event_listener)
+        self.image_generation_driver.add_event_listener(event_listener)
+        self.image_query_driver.add_event_listener(event_listener)
+        self.embedding_driver.add_event_listener(event_listener)
+        self.vector_store_driver.add_event_listener(event_listener)
```

Comment:
Potentially nice to add a method that returns all drivers in a config, then you could just iterate through them here